test/ng/parseSpec.js

```diff
@@ -2840,6 +2840,46 @@ describe('parser', function() {
             expect(filterCalled).toBe(true);
           });
 
+          it('should not be invoked unless the input/arguments change within literals', function() {
+            var filterCalls = [];
+            $filterProvider.register('foo', valueFn(function(input) {
+              filterCalls.push(input);
+              return input;
+            }));
+
+            scope.$watch('[(a | foo:b:1), undefined]');
```

Comment:
What is the point of including the params `b` and `1` here?
Should you not also be checking what happens if `b` change?

Reply:
Good question... I have no idea why I added those. I guess I'll just delete them, or if I think of a reason maybe they deserve a separate test.

Reply:
Hmm... I guess I added it because the tests above and below the new tests have it as well (which I wrote some time ago)!

I think the original reason was to ensure that constant/variables as filter arguments do not effect what is being tested. While it doesn't truely test that because it never assigns a value to the `b` variable, I think I'd vote to leave it anyway, partially just to be consistent with the other tests around it... WDYT?

Reply:
I am indifferent, it just stood out to me as something unexplained.